src/arkode/arkode_mristep.c

```diff
@@ -1285,43 +1476,94 @@ int mriStep_FullRHS(ARKodeMem ark_mem, sunrealtype t, N_Vector y, N_Vector f,
     /* compute the full RHS */
     if (!(ark_mem->fn_is_current))
     {
-      /* compute the explicit component */
-      if (step_mem->explicit_rhs)
+      /* if the method had a stiffly-accurate internal stage, use the
+         already-computed RHS vectors */
+      sa_stage = -1;
+      for (i = 1; i < step_mem->stages; i++)
```

Comment:
Yes, that's what I meant by "last row" but that wasn't very clear. My main concern was if there could be another  stiffly accurate stage in the table and we'd accidentally get the wrong row but there's not `break` in the loop so this should find the last stiffly accurate row before the embedding. However, if we're not computing the embedding, is RHS evaluated at the new state?

Reply:
Your initial suggestion worked (I didn't need a loop).  I originally added this because of the "stage groups" with MERK methods, but even those have the same assumption as the other methods, that the stage with index `step_mem->stages-1` corresponds to the overall step solution.

To answer your follow-on question, the RHS is only evaluated at the new state if the "final" stage has that "stiffly accurate" tag.  This was only needed because of a few MRI-GARK methods that *should* have had an extra all-zero row in the coefficient table, but since it was not required for fixed step sizes, that row was omitted entirely.  With the embedding, however, I had to add in that zero-row, but by labeling it as stiffly accurate and leveraging that flag here, it didn't require any extra RHS calls.

Reply:
Sorry I think my question was unclear. If you have a method with an embedding but it is being used with fixed step sizes (and the error is not being accumulated), does the RHS evaluation with the overall solution still happen or is it skipped (since we're not using the embedding)?

Reply:
I believe I figured out how to skip this (which is why all of our previous fixed-step runs did not require additional RHS evaluations in the `.out` files).

Reply:
If that RHS evaluation is skipped in that case, isn't a RHS evaluation needed here rather than a copy?